src/main/java/org/apache/flink/benchmark/full/SerializationFrameworkAllBenchmarks.java

```diff
@@ -126,6 +149,79 @@ public void serializerKryoProtobuf(FlinkEnvironmentContext context) throws Excep
 		env.execute();
 	}
 
+	/**
+	 * Source emitting an Avro GenericRecord.
+	 */
+	public static class AvroGenericRecordSource extends BaseSourceWithKeyRange<GenericRecord> implements
+			ResultTypeQueryable<GenericRecord> {
+		private static final long serialVersionUID = 2941333602938145526L;
+
+		private final GenericRecordAvroTypeInfo producedType;
+		private transient Schema myPojoSchema;
+		private final String schemaString;
+
+		private transient GenericRecord template;
+
+		public AvroGenericRecordSource(int numEvents, int numKeys, Schema schema) throws IOException {
+			super(numEvents, numKeys);
+			this.producedType = new GenericRecordAvroTypeInfo(schema);
+			this.myPojoSchema = schema;
+			this.schemaString = schema.toString();
+		}
+
+		private static Schema loadSchema() throws IOException {
+			ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+			try (InputStream is = classLoader.getResourceAsStream("avro/mypojo.avsc")) {
+				if (is == null) {
+					throw new FileNotFoundException("File 'mypojo.avsc' not found");
+				}
+				return new Schema.Parser().parse(is);
+			}
+		}
+
+		@Override
+		protected void init() throws Exception {
+			super.init();
+
+			if (myPojoSchema == null) {
+				this.myPojoSchema = new Schema.Parser().parse(schemaString);
```

Comment:
I guess the main motivation for init() to throw exception is this schema parsing. But that should actually by handled in place and translated to an IllegalStateException (since the schema was provided in the ctor and just needed to be deserialized externally).

Reply:
oh, right...that code was different when I added the exceptions. I'll remove the exceptions again since there is no use for them any more